gym/envs/robotics/hand/manipulate_touch_sensors.py

```diff
@@ -0,0 +1,146 @@
+            distance_threshold=distance_threshold, rotation_threshold=rotation_threshold, n_substeps=n_substeps, relative_control=relative_control,
+            ignore_z_target_rotation=ignore_z_target_rotation,
+        )
+        utils.EzPickle.__init__(self)
+
+        for k, v in self.sim.model._sensor_id2name.items():  # get touch sensor ids and their site names
+            if 'TS' in v:
+                # self._tsensor_name2id[v] = k
+                self._tsensor_id2name[k] = v
+                self._tsensor_id2siteid[k] = self.sim.model._site_name2id[v.replace('TS', 'T')]
+                self._site_id2intial_rgba[self._tsensor_id2siteid[k]] = self.sim.model.site_rgba[self._tsensor_id2siteid[k]].copy()  # get initial rgba values
+
+
+    def _get_obs(self):
+        robot_qpos, robot_qvel = manipulate.robot_get_obs(self.sim)
+        object_qvel = self.sim.data.get_joint_qvel('object:joint')
+        achieved_goal = self._get_achieved_goal().ravel()  # this contains the object position + rotation
+        touch_values = []  # get touch sensor readings. if there is one, set value to 1
+        if self.touch_get_obs == 'boolean':
+            for k, v in self._tsensor_id2name.items():
+                value = 1.0 if self.sim.data.sensordata[k] != 0.0 else 0.0
+                touch_values.append(value)
+        elif self.touch_get_obs == 'log':
+            for k, v in self._tsensor_id2name.items():
+                value = self.sim.data.sensordata[k]
+                touch_values.append(value)
+            if len(touch_values) > 0:
+                touch_values = np.log(np.array(touch_values) + 1.0)
+        observation = np.concatenate([robot_qpos, robot_qvel, object_qvel, touch_values, achieved_goal])
+
+        # set rgba values
+        if self.touch_visualisation == 'always':
+            for k, v in self._tsensor_id2name.items():
+                self.sim.model.site_rgba[self._tsensor_id2siteid[k]] = self._site_id2intial_rgba[self._tsensor_id2siteid[k]].copy()
+        elif self.touch_visualisation == 'on_touch':
+            for k, v in self._tsensor_id2name.items():
+                if self.sim.data.sensordata[k] != 0.0:
+                    # self.sim.model.site_rgba[self._tsensor_id2siteid[k]] = self._site_id2intial_rgba[self._tsensor_id2siteid[k]].copy()
+                    self.sim.model.site_rgba[self._tsensor_id2siteid[k]] = self.touch_color
+                else:
+                    self.sim.model.site_rgba[self._tsensor_id2siteid[k]] = self.notouch_color
+        else:
+            for k, v in self._tsensor_id2name.items():
+                self.sim.model.site_rgba[self._tsensor_id2siteid[k]] = [0, 0, 0, 0]
+
+        return {
+            'observation': observation.copy(),
+            'achieved_goal': achieved_goal.copy(),
+            'desired_goal': self.goal.ravel().copy(),
+        }
+
+    def set_object_size(self, factor):  # changes size of the manipulated object
```

Comment:
Please move this implementation to the parent class since it's equally applicable to the non-touch version.